lib/sidekiq/instrument/worker.rb

```diff
@@ -18,19 +18,17 @@ class Worker
     }.freeze
 
     def perform
-      info = Sidekiq::Stats.new
-
+      statter = Sidekiq::Stats.new
+      
       self.class::METRIC_NAMES.each do |method, stat|
-        stat ||= method
-
-        Statter.statsd.gauge("shared.sidekiq.stats.#{stat}", info.send(method))
-        Statter.dogstatsd&.gauge("sidekiq.#{stat}", info.send(method))
+        emit_stats(statter, method, stat)
       end
 
-      working = Sidekiq::Workers.new.count
-      Statter.statsd.gauge('shared.sidekiq.stats.working', working)
-      Statter.dogstatsd&.gauge('sidekiq.working', working)
+      # TODO: workers_size and working are the same, can we drop this?
+      emit_stats(statter, :workers_size, :working)
```

Comment:
If we want to keep this, why can't we move this to METRIC_NAMES hash?

Reply:
Also, I checked the Sidekiq code, but could not figure out how `Sidekiq::Stats.new.workers_size` and `Sidekiq::Workers.new.count` are the same. Could you explain why they are the same?

workers_size:
https://github.com/sidekiq/sidekiq/blob/v6.5.9/lib/sidekiq/api.rb#L123-L135

probably `Sidekiq::Workers.new.count` is based on the result of this each block.
https://github.com/sidekiq/sidekiq/blob/v6.5.9/lib/sidekiq/api.rb#L1104-L1132

Reply:
Thanks for the comments @aushm . I'll move it to draft and make corrections :)